code/datums/components/growth_and_differentiation.dm

```diff
@@ -139,3 +139,5 @@
 		old_mob.visible_message(span_warning("[old_mob] grows into \a [new_mob_name]!"))
 
 	old_mob.change_mob_type(growth_path, old_mob.loc, new_name = new_mob_name, delete_old_mob = TRUE)
+	if(initial(new_mob.unique_name))
+		old_mob.set_name()
```

Comment:
wait, we use `/mob/living/basic/spiderling/proc/grow_into_giant_spider()` as the `optional_grow_behavior` for spiderlings. shouldn't this change be there? it's good to have this regardless but i'm confused why you're touching this proc if you're concerned for spiderlings.

Reply:
also why are you invoking this proc on the old mob?